test/unit_CartesianProduct.jl

```diff
@@ -198,4 +200,14 @@ for N in [Float64, Float32, Rational{Int}]
     @test absorbing(CartesianProduct) == absorbing(CartesianProductArray) ==
           EmptySet
     @test b × e == e × b == cpa × e == e × cpa == e × e == e
+
+    # intersection
+    set1 = Hyperrectangle(low=[0.;0.], high=[1.; 1.]);
+    set2 = Hyperrectangle(low=[0.5;0.5], high=[1.5; 1.5]);
+
+    ca1 = decompose(set1, blocks=[1, dim(set1)-1])
+
+    intersect1 = intersection(ca1,set2)
```

Comment:
i don't think that this tests the interesting part in the new function, since `set2` is bounded.